boa/src/builtins/string/mod.rs

```diff
@@ -216,6 +220,46 @@ impl String {
         ))
```

Comment:
Calculating the length requires iterating over all `char`s and then we do it again to actually get the `nth` char.
This requires 2 iterations rather than just 1 that would user `unwrap_or('')`, right?
I know this isn't what you were implementing, but I noticed that it seems a bit off.

Reply:
@RageKnify No, it's O(1). Rust stores the size and returns that on `len()`.

Reply:
I'm talking about `length`, `let length = primitive_val.chars().count() as i32;`, this is O(N), is it not?

Reply:
Oh, right. I missed that part where you said it wasn't from my changes.

Reply:
@RageKnify Updated.